packages/o/opencascade/opencascade.modules.components.7.9.1.json

```
@@ -0,0 +1,1065 @@
+{
```

Can't this file be automatically generated during installation? Why does it need to be built-in? I see that gen_occt_modules_components.lua seems to generate it.

---

i don't think downloading that json would be slower than parsing the repository to create the module map. but i agree, the total size of 60 kb is too much in comparison.

for your question, it depends on the following:

- generating the json file require the occt source code/repo? can we access the source during on_load?

- or, can we define components during on_install and still have them exported to the manifest file?

the reason i asked is because i haven't been able to reliably create components when processing them outside of on_load. it seems like component metadata never gets exported to the manifest file unless it's done in the on_load hook. so using an offline json file was kind of a workaround.

i also tried getting rid of the json file by:

- globally adding components with add_components, and explicitly writing on_component per toolkit/module, with hardcoded links/deps. but xmake didn’t export them to the manifest file. when using them from client code, it takes 10+ seconds to resolve package components, often fails silently, and doesn't bring any links or proper config. components feel a bit fragile right now.

---

> i don't think downloading that json would be slower than parsing the repository to create the module map.

I want to make sure the repository git history is as light as possible. Although a single 60K file will not have much impact, if each package commit merges a 60K file, the git history of the entire xmake-repo repository will be very bloated, causing the overall git clone to slow down.

> generating the json file require the occt source code/repo? can we access the source during on_load?

If we need to parse it to add deps, we can only do it during on_load.

Since we have built-in json, why do we need to provide a lua generator?

packages/o/opencascade/gen_occt_modules_components.lua

---

Oh, I put the generator only for reproducibility and potentially reused to generate other versions (if others want to add previous versions)

I'll remove it then and make the generator private to myself, @waruqi

---

you can remove it, them add some comments to tell user how to generate it.

---

done.